cpp/include/kvikio/shim/cufile.hpp

```diff
@@ -193,39 +177,25 @@ inline bool is_cufile_available()
 }
 
 /**
- * @brief Check if cuFile's batch API is available
+ * @brief Check if cuFile's batch and stream API is available
  *
- * @return The boolean answer
- */
-#ifdef KVIKIO_CUFILE_BATCH_API_FOUND
-inline bool is_batch_available()
-{
-  try {
-    return is_cufile_available() && cuFileAPI::instance().batch_available;
-  } catch (const std::runtime_error&) {
-    return false;
-  }
-}
-#else
-constexpr bool is_batch_available() { return false; }
-#endif
-
-/**
- * @brief Check if cuFile's stream API is available
+ * Technically, the batch API is available in CUDA 12.1 but since there is no good
+ * way to check CUDA version using the driver API, we check for the existing of the
```

Comment:
What's wrong with [`cuDriverGetVersion`](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__VERSION.html#group__CUDA__VERSION_1g8b7a10395392e049006e61bcdc8ebe71)? Or do you mean check the cufile version?

Reply:
[cuDriverGetVersion](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__VERSION.html#group__CUDA__VERSION_1g8b7a10395392e049006e61bcdc8ebe71) returns the latest CUDA version supported by driver and **not** the current CUDA toolkit version :/